fix: broken links in docs (#99)
player-ui/player: docs/site/pages/architecture/cross-platform-players.mdx

---
title: 'Cross-Platform-Players'
---

# Using Player across multiple platforms

One of the major benefits of adopting Player for rendering a UI is it's ability to function across multiple platforms using the same content. In order to facilitate this, and get the most out of Player's architecture, here's a few things to keep in mind as you integrate Player into multiple platforms:

## Use core-plugins to share implementations/functionality

One of the easiest, and most beneficial thing to do when integrating Player is to organize feature sets into plugins. Keeping each plugin small and concise allows for easy sharing of features with other applications.

> The core player, as well as each of the platform integration, supplies an interface for users to augment/extend the functionality of Player. This can be used to author validations, formatters, expressions, etc once, and be shared across multiple players.

Reviewer: ```suggestion
Player Core, as well as each of the platform integration, supplies an interface for users to augment/extend the functionality of Player. This can be used to author validations, formatters, expressions, etc once, and be shared across Player implementations across platforms.
```

## What this looks like in practice

> Plugins such as the common-types plugin is a great example of a small plugin used to add a very specific feature to the core-player. Since the validations, formats, and data-types present in this plugin are authored once, we can leverage Player's architecture to share these implementations across platforms.

Reviewer: Should be a link to the common types plugin here

Author: Done

> Plugins such as the common-types plugin is a great example of a small plugin used to add a very specific feature to the core-player. Since the validations, formats, and data-types present in this plugin are authored once, we can leverage Player's architecture to share these implementations across platforms.

Reviewer: ```suggestion
Plugins such as the common-types plugin is a great example of a small plugin used to add a very specific feature to the Player Core. Since the validations, formats, and data-types present in this plugin are authored once, we can leverage Player's architecture to share these implementations across platforms.
```

> Using this mechanism, we are guaranteed that the web, iOS, and Android player configurations are able to process the same content in the same way across the systems -- features/expressions/validations added in 1 are automatically present in the other platforms without any additional work.

Reviewer: ```suggestion
Using this mechanism, we are guaranteed that the React, iOS, and Android player configurations are able to process the same content in the same way across the systems -- features/expressions/validations added in 1 are automatically present in the other platforms without any additional work.
```
